### DISCUSSION TITLE
Does `brew upgrade` respect the `-s` flag used when a formula was first installed?

### DISCUSSION THREAD

**Discussion by ywwry66** [Homebrew/discussions#2241] [Getting started] [answered]
In the `brew` man page,  it says "Upgrade outdated casks and outdated, unpinned formulae using the same options they were originally installed with". However, it seems this does not apply to `-s` option. I installed `openblas` with `brew install -s openblas`, but when I run `brew upgrade`, it still installs the bottle for the new version. Is there anything I am missing here? Thanks in advance.

**Comment by Bo98** [comment, marked as answer]
No, that's correct. "Options" here are [formula-specific options](https://docs.brew.sh/Formula-Cookbook#adding-optional-steps) that a formula may define, rather than global `brew install` options. Nothing in homebrew-core uses formula-specific options.

**Reply by ywwry66** [reply]
Thanks that makes sense. Is there an easy way to always upgrade `openblas` with source while upgrade the rest formulae with bottle?
